transcom/mymove
# [MB-16120] Prime shipment address update

## [Jira ticket](https://dp3.atlassian.net/browse/MB-16120)

## Summary

This PR builds off of the work Amanda has done [here](https://github.com/transcom/mymove/pull/10915) and includes the address update data in the response sent to the Prime. We encountered the well known `EagerPreload` [Bug](https://transcom.github.io/mymove-docs/docs/backend/setup/using-eagerpreload-in-pop/#associations-with-3-path-elements-where-the-first-2-path-elements-match) and used a workaround established in the repo.


## Verification
The primary thing to verify on this PR is that address change data is included in the response data when using the `move-task-orders/<move-id>` endpoint to fetch a move/shipment that has a address change. There are a few ways you could go about getting a move/shipment setup with address change data included but the easiest is to use the three moves that get created by devseed that have an address change: `CRQST1`, `CRQST2`,  and `CRQST3`.

1. Run `make db_dev_fresh` to update data with latest devseed data
2. Log into the prime simulator with network recording active (dev tools)
3. Search and select a move with a destination address change ( `CRQST1`, `CRQST2`,  or `CRQST3`)
4. After the move loads find the request in the network tab of dev tools. 
5. Ensure that there is `deliveryAddressUpdate` in the response from `move-task-orders/<move-id>`


### Setup to Run the Code

- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### Video

https://github.com/transcom/mymove/assets/62263329/76b37c61-2475-401a-88fd-175cf8e15e69
